### DISCUSSION TITLE
Customize Mockup Containers (e.g. Window) through Shape.

### DISCUSSION THREAD

**Discussion by netkawai** [jgraph/drawio#2663] [Q&A]
Hi

I'm using DrawIO 16.5.1. I would like to customize the Window in Mockup Container to show only one button instead of three.
![image](https://user-images.githubusercontent.com/10861481/156651400-e6d8e3ec-efb4-41d6-af93-647c96d42c68.png)

I checked Edit Diagram. The shape indicates Javascript to function, it does not Shape drawing itself.
![image](https://user-images.githubusercontent.com/10861481/156651594-87de8191-519b-4841-8be3-cd02d6e1c069.png)

I know, I can draw custom shapes through the Shape function.
https://www.diagrams.net/doc/faq/shape-complex-create-edit

However, I can find any place to get the Shape information from the existing mockups.
(I can export a mockup to SVG and edit it and import it back, but e.g. I cannot edit Window Title anymore.)

Can you help me, is there a way to edit/customize easily existing mockups?

Thank you.

**Comment by tschaffter** [comment]
I have a similar question regarding the customization of the groups (containers) provided by the library "AWS 2024" included with Draw.io. My initial goal is to create a container (listed under AWS / Groups) with a different icon. My main goal has since then switched to understanding how these containers are parametrized and work.

Consider this EC2 content group (object):

<img width="419" alt="image" src="https://github.com/user-attachments/assets/c1bab26c-525e-4cd1-961e-c2db31edd79c">

The style of this object includes the following key-values (KVs):

```
shape=mxgraph.aws4.group;grIcon=mxgraph.aws4.group_ec2_instance_contents;
```

Questions:
1. What are the values that the property `grIcon` can take? If the values are part of a set, where is this set defined? Otherwise, how can I specify a different icon, e.g. an arbitrary icon from the AWS library or an SVG image that I would have imported?
3. Where is the shape "mxgraph.aws4.group" defined?
4. What are the values that the property `shape` can take? I've seen that another value is `image` by looking at the styles of other objects. Should I be able to create my own Shape (Arrange > Insert > Shape...), give it a name (see below) and then use that name as a value for the property `shape` (e.g. `shape=my-shape`)?

```xml
<shape name="my-shape" h="100" w="100" aspect="variable" strokewidth="inherit">
  ...
</shape>
```

Thanks!

**Reply by tschaffter** [reply]
It looks like the group "mxgraph.aws4.group" is defined here: https://github.com/jgraph/drawio/blob/9c959a71542fc16787f536587947732729b0daae/src/main/webapp/js/diagramly/sidebar/Sidebar-AWS4.js#L289

Is it possible to find this "group ID" somewhere in the Draw.io web app?

**Reply by tschaffter** [reply]
@davidjgraph, do you have insights into the above questions about customizing AWS groups shipped with Draw.io?

**Reply by davidjgraph** [reply]
No, I don't.
